mysql-persistence/src/main/java/com/netflix/conductor/dao/mysql/MySQLExecutionDAO.java

```diff
@@ -805,11 +807,15 @@ private void insertOrUpdatePollData(Connection connection, PollData pollData, St
 
         if (result == 0) {
             String INSERT_POLL_DATA = "INSERT INTO poll_data (queue_name, domain, json_data) VALUES (?, ?, ?)";
+            
+            synchronized(this)
```

Comment:
This is not very efficient, can you try out upsert approach - https://dev.mysql.com/doc/refman/5.7/en/insert-on-duplicate.html?

Reply:
Let me try this approach. It seems like a good idea